packages/headless/src/controllers/case-field/headless-case-field.ts

```diff
@@ -0,0 +1,181 @@
+ */
+export interface CaseField extends Controller {
+  /**
+   * Sets the value of the specified field.
+   *
+   * @param value - The field value to set.
+   * @param updatesToFetch - A set of flags dictating whether to fetch case assist data after updating the field value.
+   */
+  update(value: string, updatesToFetch?: UpdateCaseFieldFetchOptions): void;
+
+  /**
+   * A scoped and simplified part of the headless state that is relevant to the `CaseField` controller.
+   */
+  state: CaseFieldState;
+}
+
+export interface UpdateCaseFieldFetchOptions {
+  caseClassifications?: boolean;
+  documentSuggestions?: boolean;
+}
+
+/**
+ * Creates a `CaseField` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `CaseField` controller properties.
+ * @returns A `CaseField` controller instance.
+ */
+export function buildCaseField(
+  engine: CaseAssistEngine,
+  props: CaseFieldProps = {}
+): CaseField {
+  if (!loadCaseFieldReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildController(engine);
+  const {dispatch} = engine;
+
+  const options = validateOptions(
+    engine,
+    optionsSchema,
+    props.options,
+    'buildCaseField'
+  ) as Required<CaseFieldOptions>;
+
+  dispatch(
+    registerCaseField({
+      fieldName: options.field,
+      fieldValue: '',
+    })
+  );
+
+  const getState = () => {
+    return engine.state;
+  };
+
+  return {
+    ...controller,
+
+    get state() {
+      const loading = getState().caseField?.status?.loading ?? false;
+      const error = getState().caseField?.status?.error ?? null;
+
+      const field = getState().caseField?.fields?.[options.field];
+      const value = field?.value ?? '';
+      const suggestions = field?.suggestions ?? [];
+
+      return {
+        loading,
+        error,
+        value,
+        suggestions,
+      };
+    },
+
+    update(value: string, updatesToFetch?: UpdateCaseFieldFetchOptions) {
+      const suggestionId = getState().caseField?.fields?.[
+        options.field
+      ]?.suggestions?.find((s) => s.value === value)?.id;
+
+      if (suggestionId) {
+        dispatch(logClassificationClick(suggestionId));
+      }
+
+      dispatch(
+        updateCaseField({
+          fieldName: options.field,
+          fieldValue: value,
+        })
+      );
+
+      dispatch(logUpdateCaseField(options.field));
+
+      updatesToFetch?.caseClassifications &&
+        dispatch(fetchCaseClassifications());
+      updatesToFetch?.documentSuggestions &&
+        dispatch(fetchDocumentSuggestions());
+    },
+  };
+}
+
+function loadCaseFieldReducers(
+  engine: CaseAssistEngine
+): engine is CaseAssistEngine<
+  ConfigurationSection &
+    CaseAssistConfigurationSection &
+    CaseInputSection &
+    CaseFieldSection &
+    DocumentSuggestionSection
+> {
+  engine.addReducers({configuration, caseInput, caseField, documentSuggestion});
```

Comment:
I think the `caseAssistConfiguration` reducer is missing.

Reply:
Oops. You're totally right!